felix/routetable/interface.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routetable
+
+import (
+	"github.com/projectcalico/calico/felix/ifacemonitor"
+	"github.com/projectcalico/calico/felix/ip"
+)
+
+// RouteTableSyncer is the interface used to manage data-sync of route table managers. This includes notification of
+// interface state changes, hooks to queue a full resync and apply routing updates.
+type RouteTableSyncer interface {
+	OnIfaceStateChanged(string, ifacemonitor.State)
+	QueueResync()
+	Apply() error
+}
+
+// RouteTable is the interface provided by the standard routetable module used to program the RIB.
+type RouteTableInterface interface {
```

Comment:
nit, but I think normally this would just be "RouteTable" rather than include the word "Interface" in the name.

Reply:
Yeah, I tried to avoid adding `interface` as well but there's already a RouteTable structure declared in the package. We have similar cases (e.g. IPAMConfigInterface) so I decided to do the same.